fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2463,9 +2497,19 @@ private Bundle addLinks(FHIRPagingContext context, Bundle responseBundle, String
                 log.log(Level.WARNING, "Unable to construct self link for search result bundle; using the request URI instead.", e);
             }
         }
+        requestUri = requestUri.replace("&_firstId=" + context.getFirstId(), "").replace("_firstId="
```

Comment:
I'm OK with this for now, but really we should be building an object that represents the requestUri first, then rendering that object as a uri string. Let's create a technical debt issue for that so we can avoid using string replace which has a tendency to be a little brittle.

Reply:
As suggested, a separate issue https://github.com/LinuxForHealth/FHIR/issues/4015 has been created for refactoring the request URI query parameter clean up code.